lib/membrane_live_web/channels/event_channel.ex

```diff
@@ -334,15 +334,36 @@ defmodule MembraneLiveWeb.EventChannel do
       ) do
     "event:" <> id = socket.topic
 
-    if is_ets_empty?(:main_presenters, id) do
+    main_presenter = get_main_presenter(id)
+
+    main_presenter =
+      if Presence.absent?(socket, main_presenter) do
+        remove_from_main_presenters(main_presenter, id)
+        nil
+      end
```

Comment:
I think we also have to remove him from the regular presenters list and from rtc_engine peers. LIke in `handle_in("presenter_remove"....)`